### PR TITLE
Fixes favicons fetch

### LINKED ISSUE #13281: favicons are not consistently displayed in the ledger table
### Description
@srirambv and I are not seeing favicons displayed consistently in the ledger table.

### Steps to Reproduce

  1. Add some sites to ledger table. See that sites have favicons.
  2. Delete some sites from the table.
  3. Open deleted sites dialog box.
  4. Clear one or more (or all) sites.
  5. Revisit site so it gets added back to the table.

**Actual result:**
Sites may or may not display their favicons when returned to the table. In the screenshot below all sites had been removed at one time or another (all sites had favicons prior to deleting from the table). Two of them got favicons back, but the others did not.  

screen shot 2018-02-23 at 11 57 43 am

**Expected result:**
All sites should have favicons.

**Reproduces how often:**
Most of the time, but not 100%

### Brave Version

**about:brave info:**
0.21.12

**Reproducible on current live release:**

### Additional Information
@srirambv sometimes doesn't see favicons added when sites are added to the ledger table initially.

cc @NejcZdovc

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves #13281

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).
- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review). (Ask a Brave employee to help if you cannot access this document.)

## Test Plan:
defined in #13281

Please also do regular ledger regression related to the publishers / ledger table

## Reviewer Checklist:

- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review) if one was not already requested.

Tests

- [x] Adequate test coverage exists to prevent regressions
- [x] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [x] New files have MPL2 license header

### CODE REVIEW

**Comment by NejcZdovc** [comment]
blocked on https://github.com/brave-intl/bat-publisher/pull/20

**Comment by NejcZdovc** [comment]
library updated, ready for a review

File: app/browser/api/ledger.js [hunk 1]
```diff
@@ -936,10 +947,7 @@ const fetchFavIcon = (publisherKey, url, redirects) => {
     } else if (tail > 0 && (tail + 8 >= blob.length)) return
 
     appActions.onFavIconReceived(publisherKey, blob)
-
-    if (synopsis.publishers && synopsis.publishers[publisherKey]) {
-      synopsis.publishers[publisherKey].faviconURL = blob
-    }
+    savePublisherData(publisherKey, 'faviconURL', blob)
```
Comment (bsclifton) [inline app/browser/api/ledger.js]: is this extra save needed? when the action fires, it'll also run the save

Comment (NejcZdovc) [reply app/browser/api/ledger.js]: no need

**Comment by bsclifton** [comment]
master 1f06bd2c80bb9f794b4f315f0dfd5b41b4563c26

**Pull request merged**